File: 2024/07/07/238. Product of Array Except Self/solution.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n =nums.size();
        vector<int> new_nums(n);
        new_nums[0]=1;
        int right=1;
        for(int i=1;i<n;i++){
            new_nums[i]=new_nums[i-1]*nums[i-1];
        }
        for(int j=n-1;j>=0;j--){
            new_nums[j]*=right;
            right*=nums[j];
        }
        return new_nums;
    }
};

File: 2024/07/07/238. Product of Array Except Self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result_list = [1] * len(nums)
        right = 1
        for i in range(1,len(nums)):
            result_list[i] = result_list[i-1] * nums[i-1]
        for i in range(len(nums)-1,-1,-1):
            result_list[i] *= right
            right *= nums[i]
        return result_list
            